main/src/ca/uwaterloo/flix/api/Bootstrap.scala

```diff
@@ -215,4 +435,181 @@ class Bootstrap {
+
+    // Check whether it is safe to write to the file.
+    if (Files.exists(jarFile) && !Bootstrap.isJarFile(jarFile)) {
+      throw new RuntimeException(s"The path '$jarFile' exists and is not a jar-file. Refusing to overwrite.")
+    }
+
+    // Construct a new zip file.
+    Using(new ZipOutputStream(Files.newOutputStream(jarFile))) { zip =>
+      // META-INF/MANIFEST.MF
+      val manifest =
+        """Manifest-Version: 1.0
+          |Main-Class: Main
+          |""".stripMargin
+
+      // Add manifest file.
+      Bootstrap.addToZip(zip, "META-INF/MANIFEST.MF", manifest.getBytes)
+
+      // Add all class files.
+      // Here we sort entries by relative file name to apply https://reproducible-builds.org/
+      for ((buildFile, fileNameWithSlashes) <- Bootstrap.getAllFiles(Bootstrap.getBuildDirectory(projectPath))
+        .map { path => (projectPath, Bootstrap.convertPathToRelativeFileName(Bootstrap.getBuildDirectory(projectPath), path)) }
+        .sortBy(_._2)) {
+        Bootstrap.addToZip(zip, fileNameWithSlashes, buildFile)
+      }
+    } match {
+      case Success(()) => ().toOk
+      case Failure(e) =>
+        println(e.getMessage)
+        Err(1)
+    }
+  }
+
+  /**
+    * Builds a flix package for the project.
+    */
+  def buildPkg(o: Options): Result[Unit, Int] = {
+    //
+    // We require that a Flix package has a specific structure.
+    //
+    checkProjectPath(projectPath) match {
+      case Validation.Success(_) => ()
+      case failure => throw new RuntimeException(s"Missing files or directories: ${failure.errors.mkString(", ")}")
+    }
+
+    // The path to the fpkg file.
+    val pkgFile = Bootstrap.getPkgFile(projectPath)
+
+    // Check whether it is safe to write to the file.
+    if (Files.exists(pkgFile) && !Bootstrap.isPkgFile(pkgFile)) {
+      throw new RuntimeException(s"The path '$pkgFile' exists and is not a fpkg-file. Refusing to overwrite.")
+    }
+
+    // Construct a new zip file.
+    Using(new ZipOutputStream(Files.newOutputStream(pkgFile))) { zip =>
+      // Add required resources.
+      Bootstrap.addToZip(zip, "LICENSE.md", Bootstrap.getLicenseFile(projectPath))
+      Bootstrap.addToZip(zip, "README.md", Bootstrap.getReadmeFile(projectPath))
+
+      // Add all source files.
+      // Here we sort entries by relative file name to apply https://reproducible-builds.org/
+      for ((sourceFile, fileNameWithSlashes) <- Bootstrap.getAllFiles(Bootstrap.getSourceDirectory(projectPath))
+        .map { path => (projectPath, Bootstrap.convertPathToRelativeFileName(projectPath, path)) }
+        .sortBy(_._2)) {
+        Bootstrap.addToZip(zip, fileNameWithSlashes, sourceFile)
+      }
+    } match {
+      case Success(()) => ().toOk
+      case Failure(e) =>
+        println(e.getMessage)
+        Err(1)
+    }
+  }
+
+  /**
+    * Returns a Validation containing the list of missing paths in case the path is not a project path.
+    */
+  //TODO: delete?
```

Comment:
Is this delete comment old or new?

Reply:
This is new. That is related to the other comments I left because I was unsure whether this should be deleted or not

Reply:
Can we delete it? Then lets do it

Reply:
In the issue describing this you write that it should be deleted, but I do not think this check is performed anywhere else. So is the check redundant? That is what I am unsure about.

Reply:
I see. Let's remove the check, I believe it is redundant with our new way of doing things.

(We may have to re-add a different kind of check later.)